chrome/content/zotero/xpcom/plugins.js

```diff
@@ -159,6 +159,23 @@ Zotero.Plugins = new function () {
 	};
 	
 	
+	this.getName = async function (id) {
+		var addon = await AddonManager.getAddonByID(id);
+		return addon.name;
+	};
+
+
+	/**
+	 * @param {String} id
+	 * @param {Number} idealSize In logical pixels (scaled automatically on hiDPI displays)
+	 * @returns {Promise<String | null>}
+	 */
+	this.getIconURI = async function (id, idealSize) {
+		var addon = await AddonManager.getAddonByID(id);
+		return AddonManager.getPreferredIconURL(addon, idealSize * (Zotero.hiDPI ? 2 : 1));
```

Comment:
I agree this is good enough for our purposes, but if we cared we could probably pass `Services.appShell.hiddenDOMWindow`, which has `devicePixelRatio`.

Reply:
Ah, good to know, thanks.